crates/autopilot/src/database/order_events.rs

```diff
@@ -0,0 +1,64 @@
+pub use database::order_events::OrderEventLabel;
+use {
+    chrono::{DateTime, Utc},
+    database::{
+        byte_array::ByteArray,
+        order_events::{self, OrderEvent},
+    },
+    model::order::OrderUid,
+    sqlx::PgConnection,
+    tracing::Instrument,
+};
+
+impl super::Postgres {
+    /// Stores the given order event for all passed [`OrderUid`]s in a
+    /// background task to not block the thread.
+    pub fn store_order_events(&self, uids: Vec<OrderUid>, label: OrderEventLabel) {
```

Comment:
I'd like to make this function more general and if you need a specific helper version of it to define that locally. For me the following changes make sense here:
- normal async fn
- accept `Vec<(OrderUid, Label, Timestamp)>`
- do all insertions in one transaction
- add a metric for how long the query takes like for the other DB operations

This version should be easy to convert into the slight variations needed in other places. See my comment in the solvable_orders module.

I don't like forcing the background thread because it isn't necessary. Talking to the DB should be fast and it's easier to understand the code without background thread. Especially fast when we batch all the events into one transaction as I suggest for solvable_orders.

Reply:
> accept Vec<(OrderUid, Label, Timestamp)>

The reason I didn't add that was because it wasn't necessary, yet. It appears to me the function (signature) I have at the moment is general purpose enough for the needs we have at the moment.

> do all insertions in one transaction

I wanted to avoid transactions to prevent a single failed insertion from dropping all other events.
However, I'm not sure what would cause the insertion to fail but I thought better safe than sorry. If this is a non-issue I agree that a single transaction is preferable.

Reply:
I don't see how a single event insertion can fail.

Reply:
> enough for the needs we have at the moment.

Yes but it also influences how people use the function. For example my suggestion solvable_orders might seem harder to implement because it doesn't fit the current specialized signature well. But yeah it's always a trade off to find the right amount of generalization.

Reply:
I decided to go only with `[(Uid, Label)]` because we probably never will waste a meaningful amount of time between determining labels.
[3550229](https://github.com/cowprotocol/services/pull/1641/commits/3550229b1bb49f862538032fb074ecb3ca857940)